src/bitcoin-chainstate.cpp

```diff
@@ -52,7 +53,7 @@ int main(int argc, char* argv[])
 
     // SETUP: Misc Globals
     SelectParams(CBaseChainParams::MAIN);
-    const CChainParams& chainparams = Params();
+    auto chainparams = CChainParams::Main();
```

Comment:
I don't follow why this is being changed when `SelectParams()` is still used -- `SelectParams` already relies on `gArgs`  and creates its own `unique_ptr<CChainParams>` so this is just creating two copies of the same object?

Reply:
This is a demo binary, so I think the point is to show that the `chainman` can now be instantiated with a `chainparams` that is itself instantiated without directly relying on `gArgs`. I'm not sure if this really makes sense though.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/26177#discussion_r1135671989

> This is a demo binary, so I think the point is to show that the `chainman` can now be instantiated with a `chainparams` that is itself instantiated without directly relying on `gArgs`. I'm not sure if this really makes sense though.

IMO, it would be a little clearer if the vague "SETUP: Misc Globals" comment was changed to say specifically that SelectParams() call was needed to set some globals used by validation code and could be removed when the globals were no longer needed.